File: judo-sdk/src/main/java/com/judopay/api/DateJsonDeserializer.java
package com.judopay.api;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonParseException;

import java.lang.reflect.Type;
import java.util.Date;

import static com.judopay.arch.TextUtil.isEmpty;

class DateJsonDeserializer implements JsonDeserializer<Date> {

    @Override
    public Date deserialize(final JsonElement json, final Type typeOfT, final JsonDeserializationContext context) throws JsonParseException {
        String jsonAsString = json.getAsString();
        Date date = null;
        if (!isEmpty(jsonAsString)) {
            date = Iso8601Deserializer.toDate(jsonAsString);
        }
        return date;
    }
}

File: judokit-android/src/main/java/com/judokit/android/api/deserializer/DateJsonDeserializer.kt
package com.judokit.android.api.deserializer

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import java.lang.reflect.Type
import java.util.Date

internal class DateJsonDeserializer : JsonDeserializer<Date?> {

    @Throws(JsonParseException::class)
    override fun deserialize(
        json: JsonElement,
        typeOfT: Type,
        context: JsonDeserializationContext
    ): Date? {

        val string = json.asString

        if (string.isNotEmpty()) {
            return Iso8601Deserializer.toDate(string)
        }

        return null
    }
}
